journal/src/main/java/io/camunda/zeebe/journal/file/SegmentsManager.java

```diff
@@ -215,6 +210,11 @@ Segment resetSegments(final long index) {
     }
     segments.clear();
 
+    // setting the last flushed index to a semantic-null value will let us know on start up that
+    // there is "nothing" written, even if we cannot read the descriptor (e.g. if we crash after
+    // creating the segment but before writing its descriptor)
+    metaStore.resetLastFlushedIndex();
```

Comment:
Hm, is it possible for a segment to be partially deleted? Do you mean if we crash while iterating over the segments, and only some of them are deleted? I actually don't know if resetting before would help, as we might just pick up the left over segments as is. 

What I see though is this could lead to some issues. I receive a new snapshot, but don't persist it yet - reset my log, delete the first 2 segments, crash.

On restart, I have a gap between the actual persisted snapshot (which is the old one) and first log segment. But I would probably keep those first segments? Would we even detect that?